### PR TITLE
Refactored hook handlers in mod_http_upload

### PR DESCRIPTION
**Pull request by pawlooss1**
This PR changes all hook handlers in `mod_http_upload` module to `gen_hook` format.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/148133) / small_tests / 7bdae883099b9b5f51de4b7f89e3c7bef6345c66
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3835/148133/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3835/148133/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-10-31_07.21.17/lib.mongooseim.logs/run.2022-10-31_07.21.17/suite.log.html)

File: src/http_upload/mod_http_upload.erl [hunk 1]
```diff
@@ -38,15 +38,15 @@
...
+         disco_local_items/3]).
 
 %% API
 -export([get_urls/5]).
 
 %% mongoose_module_metrics callbacks
 -export([config_metrics/1]).
 
--ignore_xref([behaviour_info/1, disco_local_items/1, process_disco_iq/5, process_iq/5]).
+-ignore_xref([behaviour_info/1, process_disco_iq/5, process_iq/5]).
```
Comment (NelsonVides) [inline src/http_upload/mod_http_upload.erl:49]: I'm actually thinking that it is very likely all these ignores can be removed 🤔

**Review by NelsonVides — COMMENTED** [review]
Just one minor comment 🤔

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/148289) / small_tests / 8d937600da77ae009a25e98239e832f8f678110b
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3835/148289/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3835/148289/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-11-02_09.21.42/lib.mongooseim.logs/run.2022-11-02_09.21.43/suite.log.html)

**Review by NelsonVides — APPROVED** [review]
LGTM 👍🏽

**Pull request merged**